src/libData/AccountData/Account.h

```diff
@@ -37,35 +38,38 @@ using namespace boost::multiprecision;
 // static OverlayDB contractStatesDB("contractStates");
 
 template<class KeyType, class DB>
-using SecureTrieDB = SpecificTrieDB<dev::HashedGenericTrieDB<DB>, KeyType>;
+using AccountTrieDB = SpecificTrieDB<dev::GenericTrieDB<DB>, KeyType>;
 
 class Account : public Serializable
 {
     boost::multiprecision::uint256_t m_balance;
     boost::multiprecision::uint256_t m_nonce;
-    h256 m_storageRoot = h256();
-    h256 m_prevRoot;
+    h256 m_storageRoot, m_prevRoot;
     h256 m_codeHash;
-
     // The associated code for this account.
+    Json::Value m_initValJson;
     vector<unsigned char> m_codeCache;
 
-    SecureTrieDB<bytesConstRef, OverlayDB> m_storage;
+    AccountTrieDB<h256, OverlayDB> m_storage;
 
 public:
     Account();
 
-    ~Account() { m_storage.init(); }
-
     /// Constructor for loading account information from a byte stream.
     Account(const vector<unsigned char>& src, unsigned int offset);
 
-    /// Constructor with account balance, and nonce.
-    Account(const uint256_t& balance, const uint256_t& nonce,
-            const h256& storageRoot, const h256& codeHash);
+    /// Constructor for a account.
+    Account(const uint256_t& balance, const uint256_t& nonce);
+
+    /// Returns true if account is a contract account
+    bool isContract() const { return m_codeHash != h256(); }
```

Comment:
`h256()` appears many times already, maybe this is not a concern in this PR but I'd like to just mention it here that we could use a global default value of `h256()` instead of constructing it every time.

Actually, a easier way to fix it inline is just declare a static variable of `h256()` here to avoid keeping construct it.